scripts/setup-mel-builddir

```diff
@@ -346,5 +346,35 @@ if [ -e $BUILDDIR/conf/local.conf ]; then
 fi
 
 if [ $toasterconfig -eq 1 ]; then
-    $layerdir/scripts/toaster_configuration "$BUILDDIR" "$MELDIR" "$BITBAKEDIR"
+    machine=$(echo "$@" | awk -F" " {'print $NF'})
+    cmdline_string="$MELDIR/meta-mentor/setup-environment "
+    check_layer_option=0
+    for i in "$@"; do
+       if [ "$i" = "-t" ]; then
+          continue
+       fi
+       if [ "$i" = "-l" ]; then
+          check_layer_option=1
+          cmdline_string="$cmdline_string $i"
+          continue
+       fi
+       if [ $check_layer_option -eq 1 ]; then
+            check_layer_option=0
+            i="\"$i\""
+            cmdline_string="$cmdline_string $i"
+            continue
+       fi
+       if [ "$i" = "$machine" ]; then
+           cmdline_string="$cmdline_string -b \"%BUILDDIR%\" $machine"
```

Comment:
Here you're adding -b $BUILDDIR when you find machine, but machine is optional. If it's not found, then you'll end up not having -b passed. Is that the intent? Have you tested this when sourcing meta-mentor/setup-environment to reconfigure an existing build directory?